File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        std::stack<int> s;

        for (const auto& i : tokens) {
            if (i == "+" || i == "-" || i == "*" || i == "/") {
                int num1 = s.top();
                s.pop();
                int num2 = s.top();
                s.pop();

                if (i == "+")
                    s.push(num2 + num1);
                if (i == "-")
                    s.push(num2 - num1);
                if (i == "*")
                    s.push(num2 * num1);
                if (i == "/")
                    s.push(num2 / num1);
            } else
                s.push(std::stoi(i));
        }

        return s.top();
    }
};

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.go
func evalRPN(tokens []string) int {
    stack := []int{}

    for _, token := range tokens{
        if token == "+" || token == "-" || token == "*" || token == "/"{
            if len(stack) <= 0 {
                return 0
            }
            i := stack[len(stack)-1]
            stack = stack[:len(stack)-1]

            j := stack[len(stack)-1]
            stack = stack[:len(stack)-1]

            if token == "+"{
                stack = append(stack, j + i)
            }
            if token == "-"{
                stack = append(stack, j - i)
            }
            if token == "*"{
                stack = append(stack, j * i)
            }
            if token == "/"{
                stack = append(stack, j / i)
            }
        } else{
            tok, err := strconv.Atoi(token)
            if err != nil{
                return 0
            }
            stack = append(stack, tok)
        }
    }

    return stack[len(stack)-1]
}